src/docs/users-guide/examples.tex

```diff
@@ -8931,3 +8929,178 @@ \subsection{Maximum Number of Steps}
+                    real[] theta,    // parameters
+                    real[] x_r,      // data (real)
+                    int[] x_i) {     // data (integer)
+  real mu = theta[1];
+  real sigma = theta[2];
+
+  return 1 / (sqrt(2 * pi()) * sigma) * exp(-0.5 * ((x - mu) / sigma)^2);
+}
+\end{stancode}
+%
+
+\noindent
+This function is expected to return the value of the integrand evaluated at point \code{x}. The
+argument \code{xc} is used in definite integrals to avoid loss of precision near
+the limits of integration and is set to NaN when either limit is infinite
+(see \refsection{integral-precision} for details on how to use this). The argument
+\code{theta} is used to pass in arguments of the integral
+that are a function of the parameters in our model. The arguments \code{x\_r}
+and \code{x\_i} are likewise used to pass in arguments of the integral that are data.
+
+\subsubsection{Strict Signature}
+%
+The function defining the integrand must have exactly the argument types and
+return type of \code{normal\_density} above, though argument naming is not important.
+Even if \code{x\_r} and \code{x\_i} are unused in the integrand, they must be
+included in the function signature. This may require passing in zero-length arrays
+for data or a zero-length vector for parameters if the integral does not involve
+data or parameters.
+
+\section{Calling the 1D Integrator}
+%
+Suppose that our model requires evaluating the lpdf of a left-truncated normal, but
+the truncation limit is to be estimated as a parameter. Because the truncation
+point is a parameter, we must include the normalization term of the truncated pdf when
+computing our model's log density. Note this is just an example of how to use the
+1D integrator. The more efficient way to perform the correct normalization in Stan
+is described in \refchapter{truncated-data}.
+
+Such a model might look like (including the function defined above):
+
+%
+\begin{stancode}
+data {
+  int N;
+  real y[N];
+}
+
+transformed data {
+  real x_r[0];
+  int x_i[0];
+}
+
+parameters {
+  real mu;
+  real<lower = 0.0> sigma;
+  real left_limit;
+}
+
+model {
+  mu ~ normal(0, 1);
+  sigma ~ normal(0, 1);
+  left_limit ~ normal(0, 1);
+  target += normal_lpdf(y | mu, sigma);
+  target += log(integrate_1d(normal_density, left_limit, positive_infinity(), { mu, sigma }, x_r, x_i));
+}
+\end{stancode}
+%
+\subsection{Limits of integration}
+The limits of integration can be finite or infinite. The infinite limits are made available via
+the Stan calls \code{negative\_infinity()} and \code{positive\_infinity()}.
+
+If both limits are either \code{negative\_infinity()} or \code{positive\_infinity()}, the integral
+and its gradients are set to zero.
+
+\subsection{Data versus Parameters}
+%
+The arguments for the real data \code{x\_r} and the integer data \code{x\_i}
+must be expressions that only involve data or transformed data variables.
+\code{theta}, on the other hand, must only involve parameters.
```

Comment:
really only parameters? No transformed parameters involving parameters and data?